article_maker.cc

```diff
@@ -642,19 +650,54 @@ void ArticleRequest::bodyFinished()
 
         closePrevSpan = true;
 
-        head += string( R"(<div class="gddictname" onclick="gdExpandArticle(')" ) + dictId + "\');"
-          + ( collapse ? "\" style=\"cursor:pointer;" : "" )
-          + "\" id=\"gddictname-" + Html::escape( dictId ) + "\""
-          + ( collapse ? string( " title=\"" ) + tr( "Expand article" ).toUtf8().data() + "\"" : "" )
-          + R"(><span class="gddicticon"><img src="gico://)" + Html::escape( dictId )
-          + R"(/dicticon.png"></span><span class="gdfromprefix">)"  +
-          Html::escape( tr( "From " ).toUtf8().data() ) + "</span><span class=\"gddicttitle\">" +
-          Html::escape( activeDict->getName().c_str() ) + "</span>"
-          + R"(<span class="collapse_expand_area"><img src="qrc:///icons/blank.png" class=")"
-          + ( collapse ? "gdexpandicon" : "gdcollapseicon" )
-          + "\" id=\"expandicon-" + Html::escape( dictId ) + "\""
-          + ( collapse ? "" : string( " title=\"" ) + tr( "Collapse article" ).toUtf8().data() + "\"" )
-          + "></span>" + "</div>";
+        // Start .gddictname
+        // Revert to QString here to be able to use its arg() method.
+        auto const dict_id_html = Html::escape( dictId );
+        QString gd_dict_name    = R"EOF(<div class="gddictname" id="gddictname-%1">)EOF";
+        gd_dict_name            = gd_dict_name.arg( dict_id_html.c_str() );
+
+        // Icon
+        gd_dict_name += R"EOF(
+        <span class="gddicticon"><img src="gico://%1/dicticon.png"></span>
+        <span class="gdfromprefix">%2</span>
+        <span class="gddicttitle">%3</span>
+        )EOF";
+        gd_dict_name = gd_dict_name.arg(
+          dict_id_html.c_str(),
+          tr( "From " ),
+          activeDict->getName().c_str()
+        );
+
+        // Collapse/expand button (blue arrow)
+        gd_dict_name += R"EOF(
+        <span class="collapse_expand_area" onclick="gdExpandArticle('%1');" >
+        <img src="qrc:///icons/blank.png" class="%2" id="expandicon-%3" title="%4">
+        </span>
+        )EOF";
+        gd_dict_name = gd_dict_name.arg(
+          dict_id_html.c_str(),
+          collapse ? "gdexpandicon" : "gdcollapseicon",
+          dict_id_html.c_str(),
+          collapse ? tr( "Expand article" ) : tr( "Collapse article" )
+        );
+
+        // If the user has enabled Anki integration in settings,
+        // Show a (+) button that lets the user add a new Anki card.
+        bool const ankiConnectEnabled = GlobalBroadcaster::instance()->getPreference()->ankiConnectServer.enabled;
+        gd_dict_name += R"EOF(
+        <a href="ankicard:%1" class="ankibutton" title="%2" style="%3" >
+        <img src="qrc:///icons/add-anki-icon.svg">
+        </a>
+        )EOF";
+        gd_dict_name = gd_dict_name.arg( dict_id_html.c_str(),
+                                         tr( "Make a new Anki note" ),
+                                         ankiConnectEnabled ? "" : "display: none;" );
+
```

Comment:
if not enabled ,I think there is no need to  output this section.

Reply:
Alright. I'll change it to a lambda since that seems like the only way to avoid the code smell warning.